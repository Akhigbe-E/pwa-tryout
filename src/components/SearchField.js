import React, { useState, useEffect } from "react";

const SearchField = ({ sendSearchInput }) => {
  const [searchInput, setsearchInput] = useState("");

  useEffect(() => {
    sendSearchInput(searchInput);
  });

  return (
    <label>
      <input
        type="search"
        placeholder="Search Robot"
        value={searchInput}
        onChange={e => {
          setsearchInput(e.target.value);
        }}
      />
    </label>
  );
};

export default SearchField;
