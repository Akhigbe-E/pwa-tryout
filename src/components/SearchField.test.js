import React from "react";
import { shallow } from "enzyme";
import SearchField from "./SearchField";

describe("SearchField Rendering", () => {
  it("render correctly", () => {
    expect(shallow(<SearchField />)).toMatchSnapshot();
  });
});

// describe("SearchField Interactions", () => {
//   it("update textfield properly", () => {
//     const wrapper = shallow(<SearchField />).find('[type="search"]');
//     wrapper.simulate("change", { target: { value: "hello" } });
//     expect(wrapper.props().value).toEqual("hello");
//   });
// });
