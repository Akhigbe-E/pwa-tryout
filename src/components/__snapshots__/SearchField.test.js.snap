// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField Rendering render correctly 1`] = `ShallowWrapper {}`;
